formats/protobuf/commonMain/src/kotlinx/serialization/protobuf/ProtoBuf.kt

```diff
@@ -234,14 +234,16 @@ class ProtoBuf(
             get() = this@ProtoBuf.context
 
         private val indexByTag: MutableMap<Int, Int> = mutableMapOf()
-        private fun findIndexByTag(desc: SerialDescriptor, serialId: Int, zeroBasedDefault: Boolean = false): Int =
-            (0 until desc.elementsCount).firstOrNull {
+        private fun findIndexByTag(desc: SerialDescriptor, serialId: Int, zeroBasedDefault: Boolean = false): Int {
+            if (desc.kind is PolymorphicKind) return serialId - 1
```

Comment:
I am really concerned about this one.
For our own polymorphic descriptor, we break our own contract in the following manner:
1) First, we declare it has 0 elements
2) Then, in `AbstractPolymorphicSerializer` we invoke `encodeX(polymorphicDescriptor, index = 0)`, `encodeX(polymorphicDescriptor, index = 1)` and same for decoding. Any sane format will break at this point and always hacking it with `if (kind == 0) treatIndexAsNonExisitng` is far from the good API design. Let's discuss it